### ISSUE TITLE
Set up tests for Ruby app

### ISSUE THREAD

**Issue by teixeirak** [ebimodeling/ghgvcR#43]
Message from @jaydorsey:

Ideally, we’d have another actual test step, where we take a few sample JSON files (generated by the Ruby app) and run them against the R code. If someone can point me to how that might work, maybe as an Rscript command, that would be great. I need to know:

- If this is possible with the current application architecture (everything is served over Rserve right now, can we still run it as a command line)
- Where I can stick the JSON files at, in the R repo (if there’s a convention)
- How to execute/trigger them (can I pass them in as a file path, does the Rscript need to include a bit about reading the file in separately, etc.)
- If there’s an easy/best practice way to manage them (testthat, wrapper script, something else, etc.)

Examples of the JSON that gets sent to the R script are found [here](ttps://github.com/rubyforgood/ghgvc/tree/master/spec/fixtures/files).

**Comment by teixeirak** [comment]
I'd like to begin by clarifying exactly what we want to do here.

There are three types of tests that I can envision:
1. Is the Ruby app passing appropriate JSON files (correct values and format) to R?
2. Given a JSON file, is the R script calculating correct values?
3. Is the app as a whole producing correct values (combined 1 & 2)?

I think its most useful to separate these processes (i.e., do tests for 1 and 2, not 3). This will facilitate identification of the sources of errors. More fundamentally, it will allow outputs of test sites to change as input parameters change. We'll still need to check input parameters, but that's simpler and more transparent than comparing GHGV calculations. 

As I understand things, then, our immediate objective should be (1)- creating tests to ensure that the JSON files passed to R are correct. We'll want to test two things:
1a. JSON files have correct values for a set of test sites
1b.  The R script successfully runs and produces values using the JSON files. 

@jaydorsey (and others), does this match your thinking?

**Comment by potterzot** [comment]
### TL;DR

There are test files in [inst/config](https://github.com/ebimodeling/ghgvcR/tree/master/inst/config). The `JSON` versions represent what should be passed to `R` from Rails. In my view, there are the following steps:

1. Tests in Rails to make sure that data passed to `R` match the format of the test files in `inst/config`.
2. Tests in R to make sure that the `inst/config` files produce correct output (partially/mostly done).
3. Tests in R to make sure the output passed back to Rails matches the format that it should
4. Tests in Rails to make sure that results received from `R` are displayed and downloadable.

In all cases, I think we need some example data files like those in `inst/config` that are kept in sync between the two repositories.

### Current testing status: 

#### 1. Is the Ruby app passing appropriate JSON files (correct values and format) to R?

We can't (I don't think?) use the `R` testthat structure to do this at package build since we need Rails involved. This might be best done on the Rails side, testing that the output conforms to the format of one of the `JSON` files in [inst/config](https://github.com/ebimodeling/ghgvcR/tree/master/inst/config).

Testing whether that output from Rails is *actually* being passed to `R` I'm not sure about either, I think that'd involve the server somehow. Regardless, this is probably not something that should get done on the `R` side, but it would be important to keep the `JSON` files used to test in sync between the two repositories.

#### 2. Given a JSON file, is the R script calculating correct values?

There are files in [inst/config/](https://github.com/ebimodeling/ghgvcR/tree/master/inst/config) that are used to test that specific functions in ghgvcR perform correctly, at least in that they don't produce errors and return the correct type of result. This mostly conforms with **However**, we still need a specific configuration and specific results to test against to make sure the answer is correct. I think we deferred this earlier but should be done.

#### Others

There's probably a 3rd section, which is that the output from `R` that goes back to Rails is in the right format, etc... If we can get a correct output file saved that matches an input file, then we can just test that the two files don't differ and that should work. Again though, we'd have to keep the two files in sync between what Rails expects back and what `R` generates to pass to Rails.
